src/dbg/debugger.cpp

```diff
@@ -35,13 +37,21 @@ struct TraceCondition
     explicit TraceCondition(String expression, duint maxCount)
         : condition(expression), steps(0), maxSteps(maxCount) {}
 
-    bool ContinueTrace()
+    inline bool ContinueTrace()
     {
         steps++;
         if(steps >= maxSteps)
             return false;
         duint value = 1;
-        return condition.Calculate(value, valuesignedcalc()) && !value;
+        if(!(condition.Calculate(value, valuesignedcalc()) && !value))
+            return false;
+        cbWatchdog(0, nullptr);

```

Comment:
Won't this horribly affect tracing performance when the watchdog is not of interest?